### PR TITLE
Remove redundant tests

### PR DESCRIPTION
**Pull request by hectorgomezv**
- Duplicated tests (service-tests already covered by controller-tests) were removed: `balances.service.spec.ts, chains.service.spec.ts, delegates.service.spec.ts, exchange.repository.spec.ts, data-decoded.repository.spec.ts, contracts.repository.spec.ts, chains.repository.spec.ts, balances.repository.spec.ts, backbone.repository.spec.ts`.
- Some tests were moved to controller tests (added validation error checks to `DelegatesController` tests).
- Some unused constructors were deleted (`CreateDelegateDto, DeleteDelegateDto`).

### CODE REVIEW

File: src/routes/delegates/entities/create-delegate.entity.ts [hunk 1]
```diff
@@ -11,18 +11,4 @@ export class CreateDelegateDto {
...
-    signature: string,
-    label: string,
-    safe?: string,
-  ) {
-    this.safe = safe;
-    this.delegate = delegate;
-    this.delegator = delegator;
-    this.signature = signature;
-    this.label = label;
-  }
```
Comment (hectorgomezv) [inline src/routes/delegates/entities/create-delegate.entity.ts:27]: Not strictly related to this PR, but the coverage report indicated this constructor was never used.

File: src/routes/delegates/entities/delete-delegate.entity.ts [hunk 2]
```diff
@@ -7,10 +7,4 @@ export class DeleteDelegateDto {
   delegator: string;
   @ApiProperty()
   signature: string;
-
-  constructor(delegate: string, delegator: string, signature: string) {
-    this.delegate = delegate;
-    this.delegator = delegator;
-    this.signature = signature;
-  }
```
Comment (hectorgomezv) [inline src/routes/delegates/entities/delete-delegate.entity.ts:15]: Not strictly related to this PR, but the coverage report indicated this constructor was never used.

File: src/routes/delegates/delegates.controller.spec.ts [hunk 3]
```diff
@@ -87,6 +87,38 @@ describe('Delegates controller', () => {
         .expect(delegatesPage);
     });
 
+    it('Should return a validation error', async () => {
```
Comment (hectorgomezv) [inline src/routes/delegates/delegates.controller.spec.ts:90]: Moving this test case to the controllers layer (as we have in other places).

**Review by fmrsabino — APPROVED** [review]
Good job 👏 This allows us to be less dependent on the implementation while still validating the correctness of the service.

**Comment by hectorgomezv** [comment]
> Good job 👏 This allows us to be less dependent on the implementation while still validating the correctness of the service.

Thanks, @fmrsabino! I had to rebase the branch, so I'd need a new approval 😄

**Review by fmrsabino — APPROVED** [review]

**Pull request merged**